app/helpers/ontology_version_helper.rb

```diff
@@ -0,0 +1,32 @@
+module OntologyVersionHelper
+  def failed?(ontology_version)
+    ontology_version.state == 'failed'
+  end
+
+  def latest_version?(ontology_version)
+    ontology_version.ontology.versions.last == ontology_version
+  end
+
+  def can_edit?(ontology, ontology_version)
+    can?(:write, ontology.repository) &&
+      failed?(ontology_version) &&
+      latest_version?(ontology_version)
+  end
+
+  def btn_url(ontology_version)
+    fancy_repository_path(resource_chain[0],
+                          path: ontology_version.path,
+                          ref: ontology_version.commit_oid,
+                          action: :show,
+                          exact_commit: !latest_version?(ontology_version))
+  end
+
+  def edit_or_view_button(ontology, ontology_version)
+    btn_text = can_edit?(ontology, ontology_version) ? 'Edit' : 'View'
+    anchor = can_edit?(ontology, ontology_version) ? '#edit' : ''
+
```

Comment:
Please cache the result of `can_edit` in a variable instead of calling it twice.